### PR TITLE
Augmented AtlasStakeStore to include an index of nodes grouped by stake;

### PR DESCRIPTION
**Pull request by spherefoundry**
# Checklist:

- [x] Code follows the style guidelines
- [x] Documentation updated
- [x] Corner cases evaluated and handled
- [x] index.js exports updated
- [x] Automatic tests added/updated
- [x] Automatic tests pass
- [x] Sanity check passed
- [x] Self-review passed
- [ ] Version updated

### CODE REVIEW

File: contracts/Storage/AtlasStakeStore.sol [hunk 1]
```diff
@@ -42,6 +44,18 @@ contract AtlasStakeStore is Base {
         return numberOfStakers;
     }
 
+    function getNumberOfStakersWithStake(uint amount) public view returns (uint32) {
```
Comment (Anze1m) [inline contracts/Storage/AtlasStakeStore.sol]: `amount` -> `initialAmount` ?

Comment (spherefoundry) [reply contracts/Storage/AtlasStakeStore.sol]: 👍

File: test/contracts/storage/atlas_stake_store.js [hunk 2]
```diff
@@ -301,4 +305,69 @@ describe('AtlasStakeStore Contract', () => {
       ).to.be.eventually.rejected;
     });
   });
+
+  describe('Index', () => {
+    const STAKE1 = 34;
+    const STAKE2 = 1267;
+
+    it('keeps track of the number of stakers', async () => {
+      const expectCount = async (all, stake1, stake2) => {
```
Comment (vanruch) [inline test/contracts/storage/atlas_stake_store.js]: `expectCorrectNumberOfStakers`?
